lib/src/modules/conditional.dart

```diff
@@ -20,7 +20,9 @@ import 'package:rohd/src/utilities/uniquifier.dart';
 /// Represents a block of logic, similar to `always` blocks in SystemVerilog.
 abstract class _Always extends Module with CustomSystemVerilog {
   /// A [List] of the [Conditional]s to execute.
-  final List<Conditional> conditionals;
+  late List<Conditional> conditionals;
+  late Logic? reset;
```

Comment:
Why do we need to save `reset` and `resetVal` here?  Also, are you even using them?

Reply:
resetVal is contains reset values for receiver logic

if the reset Logic is set and the resetVal for a receiver(Logic) 
   then resetVal gets assigned to receiver instead of Const(0)
```
if(resetVal[reveiver]):
  rec < resetVal[reveiver]
else
  rec < Const(0)
```

Reply:
I mean why do you need to declare them as class fields instead of just using the existing local variables?